### DISCUSSION TITLE
`terraform-version` manager misses new version

### DISCUSSION THREAD

**Discussion by dmaier-FN** [renovatebot/renovate#17683] [Request Help]
I'm a total newb when it comes to `renovate` config and typescript in general, but I've noticed that the `terraform` manager works fine updating our ranged `required_version` to the straight up `1.2.8` pin (which is expected), but the new-to-our-configuration `terraform-version` pin does not. As far as I know, these are the relevant parts of the output:

### `terraform` manager updates `required_version`
```json
{
  "packageFile": "myproject/production/provider.tf",
  "deps": [
    {
      "currentValue": ">= 1.0",
      "lineNumber": 15,
      "depType": "required_version",
      "datasource": "github-releases",
      "depName": "hashicorp/terraform",
      "extractVersion": "v(?<version>.*)$",
      "depIndex": 0,
      "updates": [
        {
          "updateType": "pin",
          "isPin": true,
          "newValue": "1.2.8",
          "newVersion": "1.2.8",
          "newMajor": 1,
          "branchName": "renovate/terraform-version"
        }
      ],
      "warnings": [],
      "versioning": "hashicorp",
      "sourceUrl": "https://github.com/hashicorp/terraform",
      "currentVersion": "1.2.8"
    }
  ]
}
```

### `terraform-version` does not update `.terraform/version`
```json
{
  "packageFile": ".terraform-version",
  "deps": [
    {
      "depName": "hashicorp/terraform",
      "currentValue": "1.2.7",
      "datasource": "github-releases",
      "depIndex": 0,
      "updates": [],
      "warnings": [],
      "versioning": "hashicorp",
      "sourceUrl": "https://github.com/hashicorp/terraform",
      "currentVersion": "1.2.7",
      "isSingleVersion": true,
      "fixedVersion": "1.2.7"
    }
  ]
}
```

I notice the `extractVersion` is missing in the latter, and that in the respective `lib/modules/manager` it's part of the `dep` for `terraform` and the `defaultConfig` for `terraform-version`, but I don't feel confident that that's related. Any help would be appreciated.

**Comment by rarkins** [comment]
Is it possible for you to demo this in a public repo?

**Comment by viceice** [comment]
.terraform-version` file isn't handled by `terraform` manager, it's `terraform-version` manager, so a different manager in renovate.

Also `1.2.7` is a single version, so no need for a `pin` PR. but it should have a patch update. 

Did you disable some update types?

Please provide some more config details of provide a minimal reproduction[^1]

[^1]: https://github.com/renovatebot/renovate/blob/b5ac8f04a1398ee961c5cc7a9a71018582e2bf25/docs/development/minimal-reproductions.md#L1

**Comment by henry-fn** [comment]
circling back, this was eventually fixed by changing our .terraform-version file to have a leading `v`. For example `v1.6.1` instead of `1.6.0`. This is because the terraform-version manager's extract regex hard codes the leading `v` as used in the github releases for terraform
